backends/gstreamer/lib.rs

```diff
@@ -273,6 +273,26 @@ impl Backend for GStreamerBackend {
             }
         }
     }
+
+    fn suspend(&self, id: &ClientContextId) {
+        let mut instances = self.instances.lock().unwrap();
+        match instances.get_mut(id) {
+            Some(vec) => vec.retain(|(_, weak)| {
```

Comment:
Should we abstract this operation (iterating over all known instances to perform an operation while dropping any dead ones) into a helper function that takes a closure argument, since it's used in two different places now?

Reply:
Done